### PR TITLE
Add IronDefense integration

### PR DESCRIPTION
**Pull request by marcus-defazio**
## Status
Ready

## Related Issues
N/A

## Description
Release the IronDefense Integration and example playbook.

## Screenshots
N/A

## Related PRs
N/A

branch | PR
------ | ------

## Required version of Demisto
4.5.0

## Does it break backward compatibility?
  - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [x] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
N/A

## Additional changes
Initial Commit

## Technical writer review
Mention and link to the files that require a technical writer review.
- [x] [CHANGELOG.md](https://github.com/demisto/content/pull/4591/files#diff-077dc32e993b927641c2622286782741)
- [x] [IronDefense_description.md](https://github.com/demisto/content/pull/4591/files#diff-ebec2ee71c608a6057b6261c0d9b8496)
- [x] [playbook-IronDefense_-_Example_CHANGELOG.md](https://github.com/demisto/content/pull/4591/files#diff-d3f7d51bd27c52026e1f4739f439ef99)

### CODE REVIEW

File: Integrations/IronDefense/IronDefense.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+        PARAMS = demisto.params()
+        CREDENTIALS = PARAMS.get('credentials')
+        HOST = PARAMS.get('ironAPIHost', 'localhost')
+        PORT = PARAMS.get('ironAPIPort', 443)
+        REQUEST_TIMEOUT = float(PARAMS.get('requestTimeout', 60))
+        DATA_STORE = DemistoDataStore(demisto)
+        LOGGER = DemistoLogger(demisto, LOG_PREFIX)
+
+        # initialize the IronDefense object
+        IRON_DEFENSE = IronDefense(requests.session(), HOST, PORT, CREDENTIALS, LOGGER,
```
Comment (yaakovi) [inline Integrations/IronDefense/IronDefense.py]: image

please switch to `requests.Session` as the current one is depracated
```suggestion
        IRON_DEFENSE = IronDefense(requests.Session(), HOST, PORT, CREDENTIALS, LOGGER,
```

File: Integrations/IronDefense/IronDefense.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+    def _get_error_msg_from_response(self, resp):
+        err_msg = resp.json().get('msg')
+        if err_msg is None:
+            err_msg = resp.text
+        return err_msg
+
+    '''MAIN FUNCTIONS'''
+
+    def fetch_incidents(self):
```
Comment (yaakovi) [inline Integrations/IronDefense/IronDefense.py]: was this code tested (not using unit-testing)? if not, add a comment here to specify that

File: Integrations/IronDefense/IronDefense.yml [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+- display: Username
+  name: credentials
+  required: true
+  type: 9
+- defaultvalue: '60'
+  display: Request Timeout (Sec)
+  name: requestTimeout
+  required: false
+  type: 0
+description: Integrates with IronDefense
```
Comment (yaakovi) [inline Integrations/IronDefense/IronDefense.yml]: add a more informative product description (what are the capabilities of the integration)

**Comment by yaakovi** [comment]
@kirbles19 please review the following files:
integration yml:
https://github.com/marcus-defazio/content/edit/IronDefense/Integrations/IronDefense/IronDefense.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F4591

detailed description
https://github.com/marcus-defazio/content/edit/IronDefense/Integrations/IronDefense/IronDefense_description.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F4591

**Comment by richardbluestone** [comment]
> @kirbles19 please review the following files:
> integration yml:
> https://github.com/marcus-defazio/content/edit/IronDefense/Integrations/IronDefense/IronDefense.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F4591
> 
> detailed description
> https://github.com/marcus-defazio/content/edit/IronDefense/Integrations/IronDefense/IronDefense_description.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F4591

Files have been reviewed.

**Pull request merged**